src/gui/transferlistfilterswidget.cpp

```diff
@@ -99,6 +110,62 @@ namespace
     };
 }
 
+struct FilterData
+{
+    QIcon icon;
+    QString text;
+};
+
+class ActiveFiltersLabel : public QWidget
+{
+public:
+    using QWidget::QWidget;
+
+    void setActiveFilters(const QVector<FilterData> &filters)
+    {
+        m_activeFilters = filters;
+        update();
+    };
+
+    QSize minimumSizeHint() const override
+    {
+        return QSize {0, Utils::Gui::smallIconSize(this).height()};
+    }
+
+    int heightForWidth(int) const override
+    {
+        return Utils::Gui::smallIconSize(this).height();
+    }
+
+    int activeFiltersCount() const
+    {
+        return m_activeFilters.size();
+    }
+
+private:
+    void paintEvent(QPaintEvent *) override
+    {
+        QStylePainter painter(this);
+
+        int i = 0;
+        int x = 0;
+        for (const FilterData &filter: m_activeFilters) {
+            if (i != 0) {
+                painter.drawItemText(QRect(x, 0, width() - x, height()), Qt::AlignVCenter, palette(), true, ", ");
+                x += horizontalAdvance(painter.fontMetrics(), ", ");
```

Comment:
Are lines L155, L159, and L161 compatible with Qt versions prior to Qt 5.11 as well?

Reply:
there is a check in `horizontalAdvance` function for this

Reply:
Did not catch the wrapper function, lgtm